snuba/query/snql/parser.py

```diff
@@ -1263,6 +1274,73 @@ def replace_time_condition_aliases(exp: Expression) -> Expression:
     return selector
 
 
+SORTING_VISITOR = StringifyVisitor(indenter="", delimiter="")
+
+
+def sort_key(exp: Expression) -> str:
+    return exp.accept(SORTING_VISITOR)
+
+
+def _sort_sequences(exp: Expression) -> Expression:
+    if isinstance(exp, FunctionCall) and exp.function_name in ("array", "tuple"):
```

Comment:
I am not sure you can do this across the board. 
While I hope it does not happen today, snuba allows for array and tuples columns, so you may want to compare an entire tuple with a tuple column. You cannot sort the literal in that case as the result of the comparison would change. 
An example where this could break the results is here 
https://github.com/getsentry/snuba/blob/master/snuba/query/processors/arrayjoin_keyvalue_optimizer.py#L260-L269
Now that optimization happens on the clickhouse queries so after the sorting, but relying on the fact that nobody would ever send something like that seems quite fragile.

Reply:
I could see this being done by the hierarchical hashing code, they work with that array column quite a bit. OK well this is a bust then. I'll remove this. Maybe it will still be useful without this, who knows.